### PR TITLE
Move Prometheus/OpenMetrics compatibility to its own file

### PR DESCRIPTION
**Pull request by dashpole**
This addresses https://github.com/open-telemetry/opentelemetry-specification/pull/2890#discussion_r998592184

## Changes

Move Prometheus compatibility spec from specification/metrics/data-model.md to specification/compatibility/openmetrics.md.  No other changes 

cc @jack-berg @Aneurysm9 @reyang @jsuereth @jmacd

### CODE REVIEW

File: specification/compatibility/openmetrics.md [hunk 1]
```diff
@@ -1,8 +1,369 @@
-# OpenMetrics Compatibility
+# Prometheus and OpenMetrics Compatibility
```
Comment (reyang) [inline specification/compatibility/openmetrics.md:1]: The file name says "openmetrics", consider either updating the file name to match the title or keeping the title unchanged.

**Review by reyang — APPROVED** [review]

File: specification/compatibility/openmetrics.md (hunk 1 above)
Comment (dashpole) [reply specification/compatibility/openmetrics.md]: changed to prometheus_and_openmetrics.md

File: specification/compatibility/prometheus_and_openmetrics.md [hunk 2]
```diff
@@ -0,0 +1,369 @@
+# Prometheus and OpenMetrics Compatibility
```
Comment (jsuereth) [inline specification/compatibility/prometheus_and_openmetrics.md:1]: Can you make sure there's a link to this from some other document / entry-level point?

**Review by jsuereth — APPROVED** [review]

**Review by jack-berg — APPROVED** [review]

File: specification/compatibility/prometheus_and_openmetrics.md (hunk 2 above)
Comment (dashpole) [reply specification/compatibility/prometheus_and_openmetrics.md:1]: Added in specification/README.md

**Comment by dashpole** [comment]
I removed openmetrics-guidelines.md from the semantic_conventions folder as well.   It duplicated information already present in the compatibility docs, and probably doesn't belong in the semantic conventions.

**Review by jmacd — APPROVED** [review]

**Pull request merged**